### PR TITLE
added two types of imputation #4

### LINKED ISSUE #4: Add Nan values inputer tool based on kNN.
Add to the Internal data augmenter class tool which allows to fill an using kNN approach.
https://machinelearningmastery.com/knn-imputation-for-missing-values-in-machine-learning/

### PR DESCRIPTION
**Pull request by AlexKay28**
added two types of imputation. 
maybe there are better ways of providing all possible arguments, so if u know them @joein , lets discuss it
also, imputation is possible not only for X part of X, y, but also with y. I would like to discuss the possibility to provide y inside of X and how it impacts on the effectiveness

### CODE REVIEW

File: zarnitsa/DataAugmenterInternally.py [hunk 1]
```diff
@@ -55,13 +63,34 @@ def augment_column(
         to_aug = augment_column_method[aug_type](col, **kwargs)
         return to_aug if return_only_aug else pd.concat([not_to_aug, to_aug])
 
+    def _apply_imputation_knn(self, data):
+        if self.imputer_name == "simple":
+            imputer = SimpleImputer(
+                missing_values=self.missig_values, strategy=self.imputer_strategy
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py]: typo `self.missig_values`

File: zarnitsa/DataAugmenterInternally.py [hunk 2]
```diff
@@ -9,9 +9,17 @@
...
-    def __init__(self, n_jobs=1):
+    __available_imputers = ["simple", "knn"]
+
+    def __init__(self, imputer_name=None, n_jobs=1):
+        self.imputer_name = imputer_name
         self.n_jobs = n_jobs
 
+        # this variables can be configured manually
+        self.imputer_n_neighbors = 3
+        self.missig_values = np.nan
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py]: typo `missig_values`

File: zarnitsa/DataAugmenterInternally.py [hunk 3]
```diff
@@ -55,13 +63,34 @@ def augment_column(
...
+                f"""
+                Unknown imputer name <{self.imputer_name}>.
+                Choose from {','.join(self.__available_imputers)}
+                """
+            )
+        return data
+
     def _prepare_data_to_aug(self, data, freq=0.2) -> Tuple[pd.Series, pd.Series]:
         """
         Get part of data. Not augment all of it excep case freq=1.0
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py]: typo `except`

File: zarnitsa/DataAugmenterInternally.py [hunk 4]
```diff
@@ -55,13 +63,34 @@ def augment_column(
         to_aug = augment_column_method[aug_type](col, **kwargs)
         return to_aug if return_only_aug else pd.concat([not_to_aug, to_aug])
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py]: `return_only_aug` is undefined

File: zarnitsa/DataAugmenterInternally.py [hunk 5]
```diff
@@ -55,13 +63,34 @@ def augment_column(
...
 
+    def _apply_imputation_knn(self, data):
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py]: why it is imputation_knn if we initialize here both cases - simple and knn?

File: zarnitsa/DataAugmenterInternally.py [hunk 6]
```diff
@@ -55,13 +63,34 @@ def augment_column(
...
+        if self.imputer_name == "simple":
+            imputer = SimpleImputer(
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py:69]: `SimpleImputer` undefined

File: zarnitsa/DataAugmenterInternally.py [hunk 7]
```diff
@@ -55,13 +63,34 @@ def augment_column(
...
+                missing_values=self.missig_values, strategy=self.imputer_strategy
+            )
+            data = imputer.fit_transform(data)
+        elif self.imputer_name == "knn":
+            imputer = KNNImputer(n_neighbors=self.imputer_n_neighbors)
```
Comment (joein) [inline zarnitsa/DataAugmenterInternally.py:74]: `KNNImputer` is undefined

File: zarnitsa/DataAugmenterInternally.py (hunk 4 above)
Comment (joein) [reply zarnitsa/DataAugmenterInternally.py]: `not_to_aug` is undefined too

Comment (joein) [reply zarnitsa/DataAugmenterInternally.py]: `return_only_aug` is undefined in `augment_dataframe` too

**Review by joein — CHANGES_REQUESTED** [review]

File: zarnitsa/DataAugmenterInternally.py (hunk 6 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterInternally.py:69]: my bad, fixed

File: zarnitsa/DataAugmenterInternally.py (hunk 5 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterInternally.py]: my bad, fixed

File: zarnitsa/DataAugmenterInternally.py (hunk 4 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterInternally.py]: my bad

File: zarnitsa/DataAugmenterInternally.py (hunk 1 above)
Comment (AlexKay28) [reply zarnitsa/DataAugmenterInternally.py]: fixed

**Review by joein — APPROVED** [review]

**Pull request merged**